src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,347 @@
+
+        MAXNUM = MAX
+        MINNUM = MIN
+        if (bit): # Setting minimum and maximum according to bitwise or arithmetic calculations
+            MAXNUM = BITMAX
+            MINNUM = BITMIN
+        else:
+            MAXNUM = MAX
+            MINNUM = MIN
+
+        # Generating the expression
+        for i in range(num_operands - 1):
+            self.operands[i] = random.randint(MINNUM, MAXNUM)
+            if (arithcount != 1):
+                self.operators[i] = random.randint(0, len(self.operators_list) - 1)
+            else:
+                self.operators[i] = 0
+        self.operands[num_operands - 1] = random.randint(MINNUM, MAXNUM)
+
+    # Arithmetic operations
+
+    # Addition
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < MIN:
+            op2 = MIN - random.randint(MIN, op1)
+            sum = op1 + op2
+        elif sum > MAX:
+            op2 = MAX - random.randint(op1, MAX)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
+
+    # Subtraction
+    def sub(self, op1, op2):
+        sum = op1 - op2
+        if sum < MIN:
+            op2 = - (MIN - random.randint(MIN, op1))
+            sum = op1 - op2
+        elif sum > MAX:
+            op2 = - (MAX - random.randint(op1, MAX))
+            sum = op1 - op2
+        return {'oper': '-', 'op2': op2, 'sum': sum}
+
+    # Multiplication
+    def mul(self, op1, op2):
+        sum = op1 * op2
+        if sum < MIN:
+            op2 = MIN / op1
+        elif sum > MAX:
+            op2 = MAX / op1
+        if op2 > 0:
+            op2 = math.floor(op2)
+        else:
+            op2 = math.ceil(op2)
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    # Division (floor division)
+    def div(self, op1, op2):
+        if op2 == 0:
+            if (random.randint(0,1)):
+                op2 = random.randint(1, MAX)
+            else:
+                op2 = random.randint(MIN, -1)
+        sum = op1 / op2
+        if sum > 0:
+            sum = int(math.floor(sum))
+        else :
+            sum = int(math.ceil(sum))
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    # Modulus
+    # Difference between % operator in python and js
+    # in python: -xxx % yy => yy - (abs(xxx) % yy)
+    # in js: -xxx % yy => - abs(xxx) % yy
+    def mod(self, op1, op2):
+        if op2 == 0:
+            if (random.randint(0,1)):
+                op2 = random.randint(1, MAX)
+            else:
+                op2 = random.randint(MIN, -1)
+        if (((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0)) and (op1 % op2 != 0)):
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    # Bitshifting to left
+    def bitshiftleft(self, op1, op2):
+        BITSHIFT_LEFTMAX = (2 ** 31) -1 # JS does not calculate precisely with numbers out of this interval
```

Comment:
You used the shift operator (`(1 << 53) - 1`) in MAX, please use the same here as well (`(1 << 31) - 1`)